controllers/datadoghq/watermarkpodautoscaler_controller.go

```diff
@@ -937,10 +938,10 @@ func updatePredicate(ev event.UpdateEvent) bool {
 }
 
 // SetupWithManager creates a new Watermarkpodautoscaler controller
-func (r *WatermarkPodAutoscalerReconciler) SetupWithManager(mgr ctrl.Manager, workers int) error {
+func (r *WatermarkPodAutoscalerReconciler) SetupWithManager(mgr ctrl.Manager, workers int, recoverPanic bool) error {
```

Comment:
The change appears to be unrelated and I don't think it's a change we want to have as I don't think we can guarantee everything is working as it should after encountering a `panic`.

Reply:
I split this out of this PR, but I think it's already the current behavior of the reconcilier.